### PR TITLE
feat: LogN Haste

### PR DESCRIPTION
**Pull request by thetayloredman**
**Describe your changes.**
I implemented a library `LNHaste` (depends on npm [`hastebin`](https://www.npmjs.com/package/hastebin)) that can paste to LogN's [haste.0xLogN](https://haste.0xlogn.dev) server.

This also adds a new `repoURL` property to the configuration for the user-agent.

**Additional context**

User-agent format:

```
Lexi/COMMIT7 (got/GOTVERSION; node/NODEVERSION; OS ARCH) +REPOURL
LexiProduction/COMMIT7 (got/GOTVERSION; node/NODEVERSION; OS ARCH) +REPOURL
Lexi/COMMIT7 (DirtySourceTree; got/GOTVERSION; node/NODEVERSION; OS ARCH) +REPOURL
```

Example: `Lexi/9b1fd0d (DirtySourceTree; got/5.7.1; node/v18.12.1; linux x64) +https://github.com/thetayloredman/Lexi`

Dependency changes:
```diff
+ hastebin github:thetayloredman/hastebin
```

`LexiConfig` type changes:
```diff
+ hasteServer: `${'http' | 'https'}://${string}`
+ repoURL: `${'http' | 'https'}://${string}`
```

Required changes after merge:
- Update config.ts in every node to include added `hasteServer`
- Update config.ts in every node to include added `repoURL`

TODO:
- [x] Change user agent

### CODE REVIEW

File: src/commands/admin.ts [hunk 1]
```diff
@@ -252,29 +291,8 @@ export default class AdminCommand extends LexiSlashCommand {
...
-                    case 'e':
-                    case 'error':
-                        mode = 'e';
-                        break;
-                    default:
-                        if (mode === 'i') break;
-                        await interaction.editReply({ content: 'Unknown modespec.' });
-                        return;
-                }
+                const mode = interaction.options.getString('mode')! as LogMode;
```
Comment (ilikestohack) [inline src/commands/admin.ts:294]: Does this get used anywhere?

Comment (ilikestohack) [reply src/commands/admin.ts:294]: nevermind

**Review by ilikestohack — APPROVED** [review]
Looks good

File: src/commands/admin.ts (hunk 1 above)
Comment (thetayloredman) [reply src/commands/admin.ts:294]: > Does this get used anywhere?

hmm.. what about it being the type of logs to read lmao

**Pull request merged**